sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,1014 @@
+    }
+
+    // Sentiment Runner
+    static void analyseSentimentLanguageHintRunner(BiConsumer<List<String>, String> testRunner) {
+        testRunner.accept(getSentimentInput(), "en");
+    }
+
+    static void analyseSentimentStringInputRunner(Consumer<List<String>> testRunner) {
+        testRunner.accept(getSentimentInput());
+    }
+
+    static void analyseBatchSentimentRunner(Consumer<List<TextDocumentInput>> testRunner) {
+        final List<String> sentimentInputs = getSentimentInput();
+        testRunner.accept(Arrays.asList(
+            new TextDocumentInput("0", sentimentInputs.get(0)),
+            new TextDocumentInput("1", sentimentInputs.get(1))
+        ));
+    }
+
+    void analyseBatchSentimentShowStatsRunner(
+        BiConsumer<List<TextDocumentInput>, TextAnalyticsRequestOptions> testRunner) {
+        final List<TextDocumentInput> textDocumentInputs = Arrays.asList(
+            new TextDocumentInput("0", "The hotel was dark and unclean. The restaurant had amazing gnocchi."),
+            new TextDocumentInput("1", "The restaurant had amazing gnocchi. The hotel was dark and unclean.")
+        );
+        testRunner.accept(textDocumentInputs, setTextAnalyticsRequestOptions());
+    }
+
+    static List<String> getSentimentInput() {
+        return Arrays.asList("The hotel was dark and unclean. The restaurant had amazing gnocchi.",
+            "The restaurant had amazing gnocchi. The hotel was dark and unclean.");
+    }
+
+    String getEndpoint() {
+        return interceptorManager.isPlaybackMode()
+            ? "http://localhost:8080"
+            : Configuration.getGlobalConfiguration().get("AZURE_TEXT_ANALYTICS_ENDPOINT");
+    }
+
+    /**
+     * Helper method to verify batch result.
+     *
+     * @param actualResult DocumentResultCollection<> returned by the API.
+     * @param testApi the API to test.
+     */
+    <T> void validateBatchResult(DocumentResultCollection<T> actualResult,
+        DocumentResultCollection<T> expectedResult, TestEndpoint testApi) {
+        // assert batch result
+        assertEquals(expectedResult.getModelVersion(), actualResult.getModelVersion());
+        if (this.showStatistics) {
+            validateBatchStatistics(expectedResult.getStatistics(), actualResult.getStatistics());
+        }
+        validateDocuments(expectedResult, actualResult, testApi);
+    }
+
+    /**
+     * Helper method to verify documents returned in a batch request.
+     *
+     * @param expectedResult the expected result collection..
+     * @param actualResult the actual result collection returned by the API.
+     * @param testApi the API to test.
+     */
+    private <T> void validateDocuments(DocumentResultCollection<T> expectedResult,
+        DocumentResultCollection<T> actualResult, TestEndpoint testApi) {
+        switch (testApi) {
+            case LANGUAGE:
+                final List<DetectLanguageResult> detectLanguageResults = expectedResult.stream()
```

Comment:
This is a better way to generalise your document validation. Since all your classes inherit from DocumentResult. Then any additional validations can be done on T.

```java
static void validateDetectLanguage(boolean showStatistics, DocumentResultCollection<DetectLanguageResult> expected,
    DocumentResultCollection<DetectLanguageResult> actual) {
    validateDocumentResult(showStatistics, expected, actual, (expectedItem, actualItem) -> {
        validatePrimaryLanguage(expectedItem.getPrimaryLanguage(), actualItem.getPrimaryLanguage());
        validateDetectedLanguages(expectedItem.getDetectedLanguages(), actualItem.getDetectedLanguages());
    });
}

static void validateNamedEntity(boolean showStatistics, DocumentResultCollection<NamedEntityResult> expected,
    DocumentResultCollection<NamedEntityResult> actual) {
    validateDocumentResult(showStatistics, expected, actual, (expectedItem, actualItem) -> {
        validateNamedEntities(expectedItem.getNamedEntities(), actualItem.getNamedEntities());
    });
}

private static <T extends DocumentResult> void validateDocumentResult(boolean showStatistics,
    DocumentResultCollection<T> expectedResults, DocumentResultCollection<T> actualResults,
    BiConsumer<T, T> additionalAssertions) {

    final Map<String, T> expected = expectedResults.stream().collect(Collectors.toMap(r -> r.getId(), r -> r));
    final Map<String, T> actual = actualResults.stream().collect(Collectors.toMap(r -> r.getId(), r -> r));

    assertEquals(expected.size(), actual.size());

    expected.forEach((key, expectedValue) -> {
        T actualValue = actual.get(key);

        Assertions.assertTrue(actual.containsKey(key));
        Assertions.assertNotNull(actualValue);

        if (showStatistics) {
            validateDocumentStatistics(expectedValue.getStatistics(), actualValue.getStatistics());
        }

        if (expectedValue.getError() == null) {
            assertNull(actualValue.getError());
        } else {
            assertNotNull(actualValue.getError());
            assertEquals(expectedValue.getError().getCode(), actualValue.getError().getCode());

            // Additional error validations.
        }

        additionalAssertions.accept(expectedValue, actualValue);
    });
}
```

Reply:
Will update the text framework in this issue -https://github.com/Azure/azure-sdk-for-java/issues/6952